### ISSUE TITLE
SetForegroundColour and SetBackgroundColour not working on NumCtrl

### ISSUE THREAD

**Issue by miikeat** [wxWidgets/Phoenix#808]
**Operating system**: Windows 10
**wxPython version**: 4.0.1
**Stock or custom build**: stock
**Python version**: 3.6 x64
**Stock or custom build**: Stock

**Description of the problem**:

Setting the foreground or background colour on a NumCtrl (i guess all masked Ctrls are affected) only works when done on initialization.

This works:
```python
    app=wx.App()
    frame=wx.Frame(None)
    num = NumCtrl(frame, value=0, foregroundColour='red', validBackgroundColour='blue')
    frame.Show(True)
    app.MainLoop()
```

And this not:
```python
    app=wx.App()
    frame=wx.Frame(None)
    num = NumCtrl(frame, value=0)
    num.SetForegroundColour((255,0,0))
    num.SetBackgroundColour((0,0,255))
    frame.Show(True)
    app.MainLoop()
```

**Comment by miikeat** [comment]
Seems like the internal colour variables need to be set.

This fixed it for me:
```python
from wx.lib.masked import numctrl

class NumCtrl(numctrl.NumCtrl):
    def SetForegroundColour(self, colour):
        super().SetForegroundColour(colour)
        self._foregroundColour = colour

    def SetBackgroundColour(self, colour):
        super().SetBackgroundColour(colour)
        self._validBackgroundColour = colour
```

**Comment by miikeat** [comment]
This Fixed it for me:

```python
from wx.lib.masked import numctrl

class NumCtrl(numctrl.NumCtrl):
    def SetForegroundColour(self, color):
        super().SetForegroundColour(color)
        self._foregroundColour = color

    def SetBackgroundColour(self, color):
        super().SetBackgroundColour(color)
        self._validBackgroundColour = color
```

**Comment by RobinD42** [comment]
Fixed by #869
